djangocms_link/models.py

```diff
@@ -145,6 +145,66 @@ def get_link(self):
 
         return link
 
+    def clean(self):
+        super(AbstractLink, self).clean()
+        field_names = (
+            'external_link',
+            'internal_link',
+            'mailto',
+            'phone',
+        )
+        anchor_field_name = 'anchor'
+        field_names_allowed_with_anchor = (
+            'external_link',
+            'internal_link',
+        )
+
+        cleaned_data = {
+            key: getattr(self, key)
+            for key in field_names + (anchor_field_name,)
+        }
+        anchor_field_verbose_name = force_text(
+           self._meta.get_field_by_name(anchor_field_name)[0].verbose_name)
+        anchor_field_value = cleaned_data.get(anchor_field_name)
+
+        link_fields = {
+            key: cleaned_data.get(key)
+            for key in field_names
+        }
+        link_field_verbose_names = {
+            key: force_text(self._meta.get_field_by_name(key)[0].verbose_name)
+            for key in link_fields.keys()
+        }
+        provided_link_fields = {
+            key: value
+            for key, value in link_fields.items()
+            if value
+        }
+
+        if len(provided_link_fields) > 1:
+            # Too many fields have a value.
+            error_msg = '{} {}'.format(
+                _('Only one of these fields is allowed:'),
```

Comment:
Can this be done like this:

    Only one of **xxxx**, **yyyy** or **zzzz** may be given.

"Only one of these fields is allowed" suggests that out of the fields, one of them is permissible but the others are not.